src/community/oseo/oseo-stac/src/test/java/org/geoserver/ogcapi/stac/SearchTest.java

```diff
@@ -603,4 +607,51 @@ public void testQueryByDynamicProperty() throws Exception {
         assertEquals(1, ids.size());
         assertThat(ids, contains("SAS1_20180226102021.01"));
     }
+
+    @Test
+    public void testSearchPropertySelection() throws Exception {
+        DocumentContext doc =
+                getAsJSONPath(
+                        "ogc/stac/search?fields=properties,-properties.sat:absolute_orbit,-properties.instruments,collection&filter=datetime > DATE('2017-02-25') and datetime < DATE('2017-03-31')&sortby=id",
+                        200);
+
+        checkCollectionsItemsSinglePage(
+                doc,
+                3,
+                containsInAnyOrder("SENTINEL2", "SENTINEL2", "gsTestCollection"),
+                contains(
+                        "GS_TEST_PRODUCT.01",
+                        "S2A_OPER_MSI_L1C_TL_MTI__20170308T220244_A008933_T11SLT_N02.04",
+                        "S2A_OPER_MSI_L1C_TL_SGS__20170226T171842_A008785_T32TPN_N02.04"));
+
+        JSONArray array = doc.read("features[*].properties");
+        for (int i = 0; i < array.size(); i++) {
+            Map<String, Object> props = (Map<String, Object>) array.get(i);
+            assertFalse(props.containsKey("sat:absolute_orbit"));
+            assertFalse(props.containsKey("instruments"));
+            // more then just mandatory props
+            assertTrue(props.size() > 2);
+        }
+    }
+
+    @Test
+    public void testSearchPropertySelectionStaticJsonObj() throws Exception {
+        DocumentContext doc =
+                getAsJSONPath(
+                        "ogc/stac/search?fields=properties.SENTINEL2,-properties.SENTINEL2.fullStaticObject.staticAttr1,-properties.SENTINEL2.fullStaticObject.staticAttr3.nestedStatic1&filter=datetime > DATE('2017-02-25') and datetime < DATE('2017-03-31')&sortby=id",
+                        200);
+
+        JSONArray array =
+                doc.read("features[?(@.id != 'GS_TEST_PRODUCT.01')].properties.SENTINEL2");
+        for (int i = 0; i < array.size(); i++) {
+            Map<String, Object> props = (Map<String, Object>) array.get(i);
+            Map<String, Object> staticValues = (Map<String, Object>) props.get("fullStaticObject");
+            assertFalse(staticValues.containsKey("staticAttr1"));
```

Comment:
Good, but also check the values are the correct ones.

Reply:
fixed